src/jit/codegenarmarch.cpp

```diff
@@ -2675,54 +2675,76 @@ void CodeGen::genJmpMethod(GenTree* jmp)
         if (varDsc->lvRegNum != argReg)
         {
             var_types loadType = TYP_UNDEF;
-            if (varTypeIsStruct(varDsc))
-            {
-                // Must be <= 16 bytes or else it wouldn't be passed in registers
-                noway_assert(EA_SIZE_IN_BYTES(varDsc->lvSize()) <= MAX_PASS_MULTIREG_BYTES);
-                loadType = compiler->getJitGCType(varDsc->lvGcLayout[0]);
-            }
-            else
+
+            if (varDsc->lvIsHfaRegArg())
             {
-                loadType = compiler->mangleVarArgsType(genActualType(varDsc->TypeGet()));
-            }
-            emitAttr loadSize = emitActualTypeSize(loadType);
-            getEmitter()->emitIns_R_S(ins_Load(loadType), loadSize, argReg, varNum, 0);
+                // Note that for HFA, the argument is currently marked address exposed so lvRegNum will always be
+                // REG_STK. We home the incoming HFA argument registers in the prolog. Then we'll load them back
```

Comment:
Should there be an assert that `lvRegNum == REG_STK` so that we don't miss modifying this if we ever change the implementation to enregister incoming HFAs?

Reply:
It might be useful, though duplicative. It turns out the code that precedes this loop forces arguments not in the correct register back to the stack. If they are in the correct register, it leaves them alone and then we don't do anything here. And the magic is that we always assume/assert that structs live on the stack.